modules/bpi/bpi.module

```diff
@@ -397,6 +412,116 @@ function bpi_form_node_form_alter(&$form, &$form_state, $form_id) {
 }
 
 /**
+ * Decide if the given field is a container for the given bpi image type.
+ *
+ * @param string $field_name
+ *   The field name.
+ * @param array $node
+ *   The node.
+ * @param string $image_type
+ *   The bpi image type.
+ *
+ * @return bool
+ *   True if the field is a container for the the bpi image type.
+ */
+function bpi_is_bpi_image_field($field_name, array $node, $image_type) {
+  return bpi_get_image_type($field_name, $node) == $image_type;
+}
+
+/**
+ * Place syndicated images in node edit form.
+ *
+ * @param array $form
+ *   The form.
+ * @param string $current_language
+ *   The current language.
+ * @param array $syndicated_images
+ *   The syndicated images.
+ */
+function bpi_place_images(array &$form, $current_language, array $syndicated_images) {
+  $node = $form['#node'];
+  $image_field_names = bpi_fetch_image_fields($node->type);
+  if (!empty($image_field_names)) {
+    foreach ($syndicated_images as $image_type => $images) {
+      foreach ($image_field_names as $image_field_name) {
+        if (bpi_is_bpi_image_field($image_field_name, $node, $image_type)) {
+          $field = &$form[$image_field_name][$current_language];
+          $cardinality = isset($field['#cardinality']) ? $field['#cardinality'] : 1;
```

Comment:
Thanks. Adding a comment stating that a field with an unlimited number of values has cardinality -1 would be good. This would explain the otherwise odd looking condition.